plugin/buycourses/src/buycourses.ajax.php

```diff
@@ -0,0 +1,226 @@
+<?php
+
+/* For licensing terms, see /chamilo_license.txt */
+/**
+ * Responses to AJAX calls
+ * @package chamilo.plugin.buycourses
+ */
+
+$cidReset = true;
+
+require_once '../../../main/inc/global.inc.php';
+
+api_protect_admin_script(true);
+
+$plugin = BuyCoursesPlugin::create();
+
+$paypalEnable = $plugin->get('paypal_enable');
+$comissionsEnable = $plugin->get('comissions_enable');
+
+$action = isset($_GET['a']) ? $_GET['a'] : null;
+
+switch ($action) {
+    case 'saleInfo':
+        if (api_is_anonymous()) {
+            break;
+        }
+        
+        $saleId = isset($_POST['id']) ? $_POST['id'] : '';
```

Comment:
If it's an ID, you could already filter it (intval()) unless you are absolutely sure it will be filtered later on. In this case, you should mention it in a comment. Something like this:
// $saleId is always filtered later on